private/src/gst/gstavbvideosrc.cpp

```diff
@@ -129,21 +172,50 @@ gst_avbvideosrc_change_state(GstElement *element, GstStateChange transition)
 {
     GstAvbVideoSrc *avbvideosrc = GST_AVBVIDEOSRC(element);
     GstStateChangeReturn ret = GST_STATE_CHANGE_SUCCESS;
+    GstCaps *caps_event = NULL;
 
     if (transition == GST_STATE_CHANGE_NULL_TO_READY) {
         ias_avbvideobridge_result r;
 
-        avbvideosrc->receiver = ias_avbvideobridge_create_receiver("MpegTs_Receiver",
+        avbvideosrc->receiver = ias_avbvideobridge_create_receiver("receiver",
                 avbvideosrc->stream_name);
+
```

Comment:
There's a break of locality here: you added the "caps" code between the receiver creation and check if it was created.